frmts/vrt/vrtprocesseddataset.cpp

```diff
@@ -574,6 +574,9 @@ CPLErr VRTProcessedDataset::Init(const CPLXMLNode *psTree,
             auto poBand =
                 new VRTProcessedRasterBand(this, i + 1, eOutputBandType);
             poBand->CopyCommonInfoFrom(poSrcBand);
+            poBand->SetNoDataValue(adfNoData[i]);
+            poBand->SetScale(1);
```

Comment:
I guess we would want to copy the scale and offset from `poSrcBand` if we are not using the `BandScaleOffset` algorithm, but making this behavior conditional seems messy and I'm not sure there is a use case for applying algorithms to scaled values.

Reply:
Thinking about this more, if we say (as these two lines imply) that algorithms must be run on the unscaled data, is the `BandScaleOffset` algorithm desirable at all? Or should the band scale/offset just be applied automatically when reading from the source raster?

Reply:
> Or should the band scale/offset just be applied automatically when reading from the source raster?

yes I guess applying it automatically should satisfy 99% of use cases. Do we want to provide some option in the XML to disable the automatic unscaling ?

Reply:
"unscale" attribute of "Input" with default value of "true" ? So `<Input unscale="false" />` if you want to preserve the current behavior?

Reply:
yes, that sounds good.